### ISSUE TITLE
addin error

### ISSUE THREAD

**Issue by yonicd** [r-hub/pkgsearch#84]
```r
> install.packages('pkgsearch')
trying URL 'https://cran.r-project.org/bin/macosx/el-capitan/contrib/3.6/pkgsearch_3.0.1.tgz'
Content type 'application/x-gzip' length 141660 bytes (138 KB)
==================================================
downloaded 138 KB


The downloaded binary packages are in
	/var/folders/kx/t4h_mm1910sb7vhm_gnfnx2c0000gn/T//Rtmp4mA064/downloaded_packages
> library(pkgsearch)

Error: <rlib_error in FALSE:
 The ‘memoise’, ‘shiny’, ‘shinyjs’, ‘shinyWidgets’, ‘whoami’ packages are needed for this addin.>

See `.Last.error.trace` for a stack trace.

Restarting R session...

> library(pkgsearch)

Error: <rlib_error in FALSE:
 The ‘memoise’, ‘shiny’, ‘shinyjs’, ‘shinyWidgets’, ‘whoami’ packages are needed for this addin.>

See `.Last.error.trace` for a stack trace.
```

<details closed>
<summary> <span title='Click to Expand'> session info </span> </summary>

```r

─ Session info ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 3.6.1 (2019-07-05)
 os       macOS Mojave 10.14.5        
 system   x86_64, darwin15.6.0        
 ui       RStudio                     
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       America/New_York            
 date     2019-11-27                  

─ Packages ─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 package     * version    date       lib source                        
 assertthat    0.2.1      2019-03-21 [1] CRAN (R 3.6.0)                
 backports     1.1.5      2019-10-02 [1] CRAN (R 3.6.0)                
 cli           1.1.0      2019-03-19 [1] CRAN (R 3.6.0)                
 clipr         0.7.0      2019-07-23 [1] CRAN (R 3.6.0)                
 crayon        1.3.4      2017-09-16 [1] CRAN (R 3.6.0)                
 desc          1.2.0      2019-07-25 [1] Github (r-lib/desc@c860e7b)   
 details       0.1.2      2019-11-19 [1] local                         
 digest        0.6.22     2019-10-21 [1] CRAN (R 3.6.1)                
 fastmap       1.0.0      2019-07-28 [1] CRAN (R 3.6.0)                
 htmltools     0.4.0      2019-10-04 [1] CRAN (R 3.6.0)                
 httpuv        1.5.2      2019-09-11 [1] CRAN (R 3.6.0)                
 httr          1.4.1      2019-08-05 [1] CRAN (R 3.6.0)                
 jsonlite      1.6        2018-12-07 [1] CRAN (R 3.6.0)                
 knitr         1.25       2019-09-18 [1] CRAN (R 3.6.0)                
 later         1.0.0      2019-10-04 [1] CRAN (R 3.6.0)                
 magrittr      1.5        2014-11-22 [1] CRAN (R 3.6.0)                
 memoise       1.1.0      2017-04-21 [1] CRAN (R 3.6.0)                
 mime          0.7        2019-06-11 [1] CRAN (R 3.6.0)                
 packrat       0.5.0      2018-11-14 [1] CRAN (R 3.6.0)                
 parsedate     1.2.0      2019-05-08 [1] CRAN (R 3.6.0)                
 pkgsearch   * 3.0.1      2019-11-25 [1] CRAN (R 3.6.0)                
 png           0.1-7      2013-12-03 [1] CRAN (R 3.6.0)                
 prettyunits   1.0.2      2015-07-13 [1] CRAN (R 3.6.0)                
 promises      1.1.0      2019-10-04 [1] CRAN (R 3.6.0)                
 R6            2.4.1      2019-11-12 [1] CRAN (R 3.6.0)                
 Rcpp          1.0.2      2019-07-25 [1] CRAN (R 3.6.0)                
 rlang         0.4.1      2019-10-24 [1] CRAN (R 3.6.0)                
 rprojroot     1.3-2      2018-01-03 [1] CRAN (R 3.6.0)                
 rstudioapi    0.10       2019-03-19 [1] CRAN (R 3.6.0)                
 sessioninfo   1.1.1      2018-11-05 [1] CRAN (R 3.6.0)                
 shiny         1.3.2.9001 2019-08-28 [1] Github (rstudio/shiny@d947814)
 shinyjs       1.0        2018-01-08 [1] CRAN (R 3.6.0)                
 whoami        1.3.0      2019-03-19 [1] CRAN (R 3.6.0)                
 withr         2.1.2      2018-03-15 [1] CRAN (R 3.6.0)                
 xfun          0.10       2019-10-01 [1] CRAN (R 3.6.0)                
 xml2          1.2.2      2019-08-09 [1] CRAN (R 3.6.0)                
 xtable        1.8-4      2019-04-21 [1] CRAN (R 3.6.0)                

[1] /Library/Frameworks/R.framework/Versions/3.6/Resources/library

```

</details>
<br>

**Comment by gaborcsardi** [comment]
That is weird. This error is from `pkg_search_addin()`, but why is that called automatically? Can you please try without RStudio, and if it works in the terminal then send us your RStudio version? Thanks.

**Comment by yonicd** [comment]
same in the cli

```bash
Yonis-MacBook-Pro:~ yonis$ R

R version 3.6.1 (2019-07-05) -- "Action of the Toes"
Copyright (C) 2019 The R Foundation for Statistical Computing
Platform: x86_64-apple-darwin15.6.0 (64-bit)
```

```r
R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> library(pkgsearch)
> pkg_search_addin()

Error: <rlib_error in FALSE:
 The ‘memoise’, ‘shiny’, ‘shinyjs’, ‘shinyWidgets’, ‘whoami’ packages are needed for this addin.>

See `.Last.error.trace` for a stack trace.```

**Comment by gaborcsardi** [comment]
Can you send the output of `.Last.error.trace` after the error happens?

**Comment by gaborcsardi** [comment]
Oh, wait, that happens after you call `pkg_search_addin()`? Then you need to install these packages to use the addin.

I thought it happened after `library(pkgsearch)` like you had it in your original post.

**Comment by yonicd** [comment]
they are installed (see sessioninfo in original comment)

```> .Last.error.trace

 Stack trace:

 1. pkgsearch:::pkg_search_addin()
 2. pkgsearch:::needs_packages(c("memoise", "shiny", "shinyjs", "shinyWidgets", ...
 3. throw(new_error("The ", paste(sQuote(pkgs), collapse = ", "),  ...

 x The ‘memoise’, ‘shiny’, ‘shinyjs’, ‘shinyWidgets’, ‘whoami’ packages are needed for this addin. ```

**Comment by yonicd** [comment]
it was `shinyWidgets`. that error from `needs_packages` is misleading, can it return only the packages that need to be installed?

**Comment by maelle** [comment]
fixed in #86
